Fix new warning introduced by std::vector<bool> serializer overload

```
In file included from /home/jliffla/vt-auto-build/vt-script-build-1/checkpoint/checkpoint/src/serdes_headers.h:58,
                 from /home/jliffla/vt-auto-build/vt-script-build-1/checkpoint/checkpoint/src/serialization_library_headers.h:48,
                 from /home/jliffla/vt-auto-build/vt-script-build-1/checkpoint/checkpoint/tests/unit/test_vector_bool_serialize.cc:47:
/home/jliffla/vt-auto-build/vt-script-build-1/checkpoint/checkpoint/src/container/vector_serialize.h: In instantiation of ‘void serdes::serialize(Serializer&, std::vector<bool, VectorAllocator>&) [with Serializer = serdes::Sizer; VectorAllocator = std::allocator<bool>]’:
/home/jliffla/vt-auto-build/vt-script-build-1/checkpoint/checkpoint/src/dispatch/dispatch_serializer_nonbyte.h:238:16:   required from ‘void serdes::SerializerDispatchNonByte<SerializerT, T>::apply(SerializerT&, T*, serdes::SerialSizeType, serdes::SerializerDispatchNonByte<SerializerT, T>::hasNoninSerialize<U>*) [with U = std::vector<bool>; SerializerT = serdes::Sizer; T = std::vector<bool>; serdes::SerialSizeType = long unsigned int; serdes::SerializerDispatchNonByte<SerializerT, T>::hasNoninSerialize<U> = std::vector<bool>]’
/home/jliffla/vt-auto-build/vt-script-build-1/checkpoint/checkpoint/src/dispatch/dispatch_serializer_nonbyte.h:161:29:   required from ‘void serdes::SerializerDispatchNonByte<SerializerT, T>::operator()(SerializerT&, T*, serdes::SerialSizeType) [with U = std::vector<bool>; SerializerT = serdes::Sizer; T = std::vector<bool>; serdes::SerialSizeType = long unsigned int]’
/home/jliffla/vt-auto-build/vt-script-build-1/checkpoint/checkpoint/src/dispatch/dispatch_serializer_byte.h:108:13:   required from ‘void serdes::SerializerDispatchByte<SerializerT, T>::operator()(SerializerT&, T*, serdes::SerialSizeType, serdes::SerializerDispatchByte<SerializerT, T>::isNotByteCopyType<U>*) [with U = std::vector<bool>; SerializerT = serdes::Sizer; T = std::vector<bool>; serdes::SerialSizeType = long unsigned int; serdes::SerializerDispatchByte<SerializerT, T>::isNotByteCopyType<U> = std::vector<bool>]’
/home/jliffla/vt-auto-build/vt-script-build-1/checkpoint/checkpoint/src/dispatch/dispatch.impl.h:61:5:   required from ‘static serdes::SerialSizeType serdes::Dispatch<T>::sizeType(T&) [with T = std::vector<bool>; serdes::SerialSizeType = long unsigned int]’
/home/jliffla/vt-auto-build/vt-script-build-1/checkpoint/checkpoint/src/dispatch/dispatch.impl.h:217:46:   required from ‘serdes::SerializedReturnType serdes::serializeType(T&, serdes::BufferObtainFnType) [with T = std::vector<bool>; serdes::SerializedReturnType = std::tuple<std::unique_ptr<serdes::Buffer, std::default_delete<serdes::Buffer> >, long unsigned int>; serdes::BufferObtainFnType = std::function<char*(long unsigned int)>]’
/home/jliffla/vt-auto-build/vt-script-build-1/checkpoint/checkpoint/src/serialize_interface.impl.h:59:40:   required from ‘serialization::interface::SerializedReturnType serialization::interface::serialize(T&, serialization::interface::BufferCallbackType) [with T = std::vector<bool>; serialization::interface::SerializedReturnType = std::unique_ptr<serialization::interface::SerializedInfo>; serialization::interface::BufferCallbackType = std::function<char*(long unsigned int)>]’
/home/jliffla/vt-auto-build/vt-script-build-1/checkpoint/checkpoint/tests/unit/test_vector_bool_serialize.cc:54:81:   required from here
/home/jliffla/vt-auto-build/vt-script-build-1/checkpoint/checkpoint/src/container/vector_serialize.h:77:24: warning: comparison of integer expressions of different signedness: ‘int’ and ‘std::vector<bool>::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]
     for (auto i = 0; i < vec.size(); ++i) {
```

--- a/src/container/vector_serialize.h
+++ b/src/container/vector_serialize.h
@@ -74,7 +74,7 @@ void serialize(Serializer& s, std::vector<bool, VectorAllocator>& vec) {
       s | elt;
     }
   } else {
-    for (auto i = 0; i < vec.size(); ++i) {
+    for (size_t i = 0; i < vec.size(); ++i) {
       bool elt;
       s | elt;
       vec[i] = elt;

